src/zope/i18n/tests/test_formats.py

```diff
@@ -171,19 +187,23 @@ def testParseQuotesInPattern(self):
 
     def testParseDateTimePatternError(self):
         # Quote not closed
-        try:
+        with self.assertRaisesRegex(
+                DateTimePatternParseError,
+                'The quote starting at character 2 is not closed.'):
             parseDateTimePattern("HH' Uhr")
-        except DateTimePatternParseError as err:
-            self.assertEqual(
-                str(err), 'The quote starting at character 2 is not closed.')
+
         # Test correct length of characters in datetime fields
-        try:
-            parseDateTimePattern("HHHHH")
-        except DateTimePatternParseError as err:
-            self.assertTrue(str(err).endswith('You have: 5'))
+        parseDateTimePattern("HHHHH")
```

Comment:
Uh, I think this was supposed to fail.  Why didn't it fail?

Reply:
Good question. There's nothing in the code that produces an error message like that. The parsing function is actually quite lenient and will basically only ever produce one error (for unclosed quotes).